helix-term/src/ui/editor.rs

```diff
@@ -523,6 +523,9 @@ impl EditorView {
                 // if this fails, count was Some(0)
                 // debug_assert!(cxt.count != 0);
 
+                // set the register
+                cxt.register = cxt.editor.register.take().unwrap_or('"');
```

Comment:
We shouldn't default to `"`. I don't think by default we need to yank to system register.

Reply:
`"` seems to be just the general register in kakoune (https://github.com/mawww/kakoune/blob/master/doc/pages/registers.asciidoc#default-registers), we're using `+` for the clipboard

Reply:
If it's the default why don't we just make it `char` rather than `Option<char>`?

Reply:
Similar to how `count` is handled (even though default is `1`), it indicates that a value has been explicitly set with `"` command. Behavior is pretty much the same though.

Reply:
We could do `mem::replace` which is a single line either.

Reply:
I do prefer the explicit `Option` on both `Editor` and the `Context` so we know if a register was explicitly set or not: that's one of the issues with `count` currently: it uses an Option on the `Editor`, but it's an `int` on the `Context` so there's problems with `1g`: https://github.com/helix-editor/helix/blob/8c2fa12ffc006de787992b4022263be033ac8200/helix-term/src/commands.rs#L1286-L1287

Reply:
Maybe I should use `Option<char>` on `Context` as well then?

Reply:
Yep!

Reply:
> I do prefer the explicit Option on both Editor and the Context so we know if a register was explicitly set or not: that's one of the issues with count currently: it uses an Option on the Editor, but it's an int on the Context so there's problems with 1g: 

Can we do `Option<NonZeroUsize>` for that? We can also be sure `0` won't be valid.